hnn_core/extracellular.py

```diff
@@ -28,6 +28,30 @@
 from .externals.mne import _validate_type, _check_option
 
 
+def calculate_csd2d(data, delta=1):
+    """current source density (csd) estimation
+
+    The three-point finite-difference approximation of the 
+    second spatial derivative for computing 1-dimensional CSD
+    Parameters
+    ----------
+    data : channels x times array
+        LFP data
+    delta : int
+        distance between channels (in um), scales the CSD
+    Returns
+    -------
+    csd2d : channels x times array
+        the 2nd derivative current source density estimate (csd2d)
+    """
+    csd2d = np.zeros([data.shape[0] - 2, data.shape[1]])
+
+    for channel, trace in enumerate(data[1:-1], start=1):
```

Comment:
any index variable call it `xyz_idx`. Here it would be `ch_idx`

Reply:
Also, there might be a faster and simpler way to write this using [numpy.diff](https://numpy.org/doc/stable/reference/generated/numpy.diff.html).